src/seedu/addressbook/Main.java

```diff
@@ -111,6 +111,9 @@ private CommandResult executeCommand(Command command)  {
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            ui.showStorageReadOnlyMessage();
+            return new CommandResult(e.getMessage());
```

Comment:
Although this exception message is pretty friendly, you should avoid using the exception message directly. Also notice that 2 messages are printed in this case